synapse/appservice/scheduler.py

```diff
@@ -59,6 +59,8 @@
 
 logger = logging.getLogger(__name__)
 
+MAX_EVENTS_PER_TRANSACTION = 100
```

Comment:
It's a bit confusing that this means a transaction will actually send 200 events - 100 ephemeral and 100 persistent. Perhaps pop a comment above this saying that this limit is applied to both ephemeral and persistent event types.